### PR TITLE
Improvements for SizeMappingV2 (advanced size mapping) module

### PR DESCRIPTION
**Pull request by Fawke**
## Type of change

- [X] Enhancements
- [X] Bugfix
- [X] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)

## Description of change

With recent addition of '[Identical Ad Units](https://github.com/prebid/Prebid.js/pull/5062)' feature to this module, some log messages getting displayed were not very informative. Refer to [this](https://github.com/prebid/Prebid.js/issues/5103) issue for more details. 

- This PR implements the potential solution discussed in that issue.

- Log messages have been improved for Size Mapping V2 for more clarity when debugging.

- The example setup in `integrationExamples/gpt/advanced_size_mapping.html` now has instructive comments to help the user understand what's going on. 

- Resolves a minor bug in which bidder level sizeConfig checks were not checking one specific scenario.

Resolves https://github.com/prebid/Prebid.js/issues/5103

### CODE REVIEW

File: modules/sizeMappingV2.js [hunk 1]
```diff
@@ -429,57 +438,56 @@ export function getRelevantMediaTypesForBidder(sizeConfig, activeViewport) {
...
   // check if the adUnit exists already in the sizeMappingInterStore (check for equivalence of 'code' && 'mediaTypes' properties)
   const adUnitDetail = adUnitsForAuction.filter(adUnitDetail => adUnitDetail.adUnitCode === adUnit.code && utils.deepEqual(adUnitDetail.mediaTypes, adUnit.mediaTypes));
 
   if (adUnitDetail.length > 0) {
+    adUnitDetail[0].cacheHits++;
     return adUnitDetail[0];
   } else {
-    const { mediaTypes, sizeBucketToSizeMap, activeViewport, transformedMediaTypes } = internal.getFilteredMediaTypes(adUnit.mediaTypes);
+    const identicalAdUnit = adUnitsForAuction.filter(adUnitDetail => adUnitDetail.adUnitCode === adUnit.code);
+    const adUnitInstance = identicalAdUnit.length > 0 ? typeof identicalAdUnit[0].instance === 'number' ? identicalAdUnit[identicalAdUnit.length - 1].instance + 1 : 1 : 1;
```
Comment (msm0504) [inline modules/sizeMappingV2.js:453]: Are the nested ternary operators needed here? Can this be rewritten as:
```
identicalAdUnit.length > 0 && typeof identicalAdUnit[0].instance === 'number' ? identicalAdUnit[identicalAdUnit.length - 1].instance + 1 : 1;
```

File: modules/sizeMappingV2.js [hunk 2]
```diff
@@ -429,57 +438,56 @@ export function getRelevantMediaTypesForBidder(sizeConfig, activeViewport) {
...
+    const isLabelActivated = internal.isLabelActivated(adUnit, labels, adUnit.code, adUnitInstance);
+    const { mediaTypes = adUnit.mediaTypes, sizeBucketToSizeMap, activeViewport, transformedMediaTypes } = isLabelActivated && internal.getFilteredMediaTypes(adUnit.mediaTypes);
```
Comment (msm0504) [inline modules/sizeMappingV2.js:455]: If `isLabelActivated ` is false, then all of the destructured properties will be `undefined`. Is that correct? Could that cause any errors elsewhere?

Comment (Fawke) [reply modules/sizeMappingV2.js:455]: Yes, those properties can be left as undefined if `isLabelActivated` is false. Only exception being `medaiTypes` which I've already initialized.

This step is done for optimization. Basically the idea is, if an entire ad unit is deactivated by a failing label check, there is no need call function `getFilteredMediaTypes` and compute those values which anyway won't be used.

File: modules/sizeMappingV2.js (hunk 1 above)
Comment (Fawke) [reply modules/sizeMappingV2.js]: @msm0504 Good catch! I've made the necessary change.

**Pull request merged**